### PR TITLE
Newtodo

### LINKED ISSUE #22116: ivy.<lu_solve>
**Links to native framework implementations**
- [TensorFlow](<[FUNC_URL_DOCS](https://www.tensorflow.org/api_docs/python/tf/linalg/lu_solve)>)

### PR DESCRIPTION
**Pull request by TomiwaPhilip**
Close #22116

### CODE REVIEW

**Review by hello-fri-end — COMMENTED** [review]
Hey @TomiwaPhilip! Thanks for opening this PR. It seems the issue that you have linked with your PR isn't tracked by any ToDo list issue. Maybe you forgot to comment the issue no. in the respective ToDo list issue. Could you please look into this?

**Comment by TomiwaPhilip** [comment]
@hello-fri-end

Hi! Thanks for the tip. I have corrected the error. It is now linked to the todo list: tensorflow frontend linear algebra issue.
Is there anything more I should do to get this pull request accepted?

**Comment by TomiwaPhilip** [comment]
@hello-fri-end 

Please review this PR and merge if there are no more changes to effect!

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -470,3 +470,8 @@ def tensor_diag_part(input, name=None):
     reshaped = ivy.reshape(input, (prod, prod))
     diagonal = ivy.diagonal(reshaped)
     return ivy.reshape(diagonal, tuple(half_shape))
+
+
+@to_ivy_arrays_and_back
+def lu_solve(lower_upper, rhs):  # ToDO
```
Comment (hello-fri-end) [inline ivy/functional/frontends/tensorflow/linalg.py:476]: The function signature should match with `tf.linalg.lu_solve` function :
```python
tf.linalg.lu_solve(
    lower_upper, perm, rhs, validate_args=False, name=None)

```

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 2]
```diff
@@ -470,3 +470,8 @@ def tensor_diag_part(input, name=None):
...
+    return ivy.solve(lower_upper=lower_upper, rhs=rhs)
```
Comment (hello-fri-end) [inline ivy/functional/frontends/tensorflow/linalg.py:477]: `ivy.solve` function does not have any keyword arguments `lower_upper` and `rhs` :
https://unify.ai/docs/ivy/docs/functional/ivy/linear_algebra/ivy.functional.ivy.linear_algebra.solve.html#solve

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @TomiwaPhilip ! Thanks for linking the issue. Would you please look at the documentation of `tf.linalg.lu_solve` and update your implementation based on the comments I have left now. Feel free to request another review when you are done with the changes. Thanks

File: ivy/functional/frontends/tensorflow/linalg.py (hunk 2 above)
Comment (TomiwaPhilip) [reply ivy/functional/frontends/tensorflow/linalg.py:477]: @hello-fri-end 
There is no function on ivy documentation that matches tf.linalg.lu_solve that is why I was instructed by one of Ivy reviewers to implement it compositionally with a present ivy function. That is what I am trying to do here.  And that is why I added #Todo as a comment in the code.

Do you understand, sir?

Comment (hello-fri-end) [reply ivy/functional/frontends/tensorflow/linalg.py:477]: Hey @TomiwaPhilip, you are right that the function should be implemented compositionally - this means you should use existing ivy functions to replicate the behavior of the function you're implementing. However, it seems the way you have tried to implement this function doesn't match with the functionality of `tf.linalg.lu_solve` ( look at the source code here ; https://github.com/tensorflow/tensorflow/blob/v2.13.0/tensorflow/python/ops/linalg/linalg_impl.py#L937-L1032). Here's a blog post where you can understand how `tf.linalg.lu_solve` works -> https://biswajitsahoo1111.github.io/post/doing-linear-algebra-using-tensorflow-2/. Once you have understood what the native function is supposed to do, you can easily replicate the behavior using ivy functions in your implementation. Happy coding :)

**Pull request closed without merging**